crates/c-api/src/types/memory.rs

```diff
@@ -71,20 +73,28 @@ pub extern "C" fn wasmtime_memorytype_new(
     maximum_specified: bool,
     maximum: u64,
     memory64: bool,
+    shared: bool,
 ) -> Box<wasm_memorytype_t> {
     let maximum = if maximum_specified {
         Some(maximum)
     } else {
         None
     };
-    Box::new(wasm_memorytype_t::new(if memory64 {
-        MemoryType::new64(minimum, maximum)
-    } else {
-        MemoryType::new(
+    Box::new(if shared {
```

Comment:
Mind panicking here for now with `unimplemented!()` if both `shared` and `memory64` are set? (we need to add a constructor for that but haven't gotten around to it yet)

Reply:
Actually I take this back, this may want to use `MemoryTypeBuilder` to support shared 64-bit memories